### ISSUE TITLE
Running into issue port already in use

### ISSUE THREAD

**Issue by dzincone** [JungleCatSW/reactioncommerce-aws-elasticbeanstalk#1]
INFO: Docker container 1f741df9d2b7 is running aws_beanstalk/current-app.
ERROR: [Instance: i-0cccb2bac7313d483] Command failed on instance. Return code: 1 Output: (TRUNCATED)...[emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
nginx: [emerg] still could not bind()
[FAILED]. 

This happens from editing the file and running the bash script for redirects. Any idea on how to fix that?

**Comment by strawbs1** [comment]
When does this happen ? from a fresh startup ? Can you find out what is running on that port ?

**Comment by Gary-Pan** [comment]
I have the same issue from a fresh startup.

Application deployment failed at 2017-10-26T05:58:56Z with exit status 1 and error: Hook /opt/elasticbeanstalk/hooks/appdeploy/post/99_restart_delayed_job.sh failed.

Stopping nginx: [ OK ] 
Starting nginx: nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
nginx: [emerg] still could not bind()
[FAILED].


Then I check the 80 port on that instance as follow.
$ sudo lsof -i :80 
COMMAND  PID  USER   FD   TYPE DEVICE SIZE/OFF NODE NAME
nginx   4573  root    6u  IPv4  15742      0t0  TCP *:http (LISTEN)
nginx   4576 nginx    6u  IPv4  15742      0t0  TCP *:http (LISTEN)

**Comment by strawbs1** [comment]
Seems like nginx was already running ? are you using the setup below ?
64bit Amazon Linux 2016.09 v2.5.2 running Docker 1.12.6

**Comment by Gary-Pan** [comment]
Yes, nginx was already running.
I used the default platform 64bit Amazon Linux 2017.03 v2.7.4 running Docker 17.03.2-ce

**Comment by hfmw** [comment]
Hi @Gary-Pan @JungleCatSW @dzincone ,
I'm getting this too. How did you fix it?
Cheers!
Will

**Comment by hfmw** [comment]
For anyone reading this there is an issue with nginx-setup-config. It works when you first deploy, but if you modify setup or redeploy it gives errors. I think this is due to the folder move commands at the bottom of the file.

**Comment by strawbs1** [comment]
Hi @willmoss1000 the script is designed for blue-green deployment. You shouldn't reuse an EB instance, instead start a new one with your modifications and then transfer the the urls once the new EB is ready. Then terminate the original EB. Use the menu option highlighted in red below to swap urls:
![blue-green](https://user-images.githubusercontent.com/14792977/37875400-5d3168f4-3036-11e8-92b2-a4a88df8d7ad.png)

This way your site has no downtime!

Look up 'Blue Green' Deployment to find out more.

**Comment by hfmw** [comment]
Ah I see, thanks @JungleCatSW!

Sorry... just one more question. I'm getting the following error: Error during WebSocket handshake: Unexpected response code: 400. This must be something to do with the nginx config? Why might this be? 

Thanks!!

**Comment by strawbs1** [comment]
@willmoss1000 Hi if you are still getting this can you start another issue on here and paste in the full error and where the error log came from. Cheers!

**Comment by hfmw** [comment]
Thanks @JungleCatSW! So like the others I'm running into this error: 

/var/log/nginx/error.log
2018/03/27 11:59:13 [emerg] 4828#0: bind() to 0.0.0.0:80 failed (98: Address already in use)
2018/03/27 11:59:13 [emerg] 4828#0: bind() to 0.0.0.0:80 failed (98: Address already in use)
2018/03/27 11:59:13 [emerg] 4828#0: bind() to 0.0.0.0:80 failed (98: Address already in use)
2018/03/27 11:59:13 [emerg] 4828#0: bind() to 0.0.0.0:80 failed (98: Address already in use)
2018/03/27 11:59:13 [emerg] 4828#0: bind() to 0.0.0.0:80 failed (98: Address already in use)
2018/03/27 11:59:13 [emerg] 4828#0: still could not bind()
2018/03/27 11:59:19 [error] 4817#0: *1 connect() failed (111: Connection refused) while connecting to upstream, client: XXXXX, server: www.XXXXXXX, request: "GET / HTTP/1.1", upstream

Even if I fix it with sudo fuser -k 80/tcp I get an error saying that proxy isn't working.

I'm pretty sure I've followed your steps exactly. What do can I do to prevent this error on startup?

Cheers!

**Comment by strawbs1** [comment]
@willmoss1000 ill take a look and get back to you

**Comment by strawbs1** [comment]
@willmoss1000 the 400 error is different to the port in use error though right ?
The port in use error is just because scripts shouldnt be reused.
A 400 error is a bad request, so it could be to do with the domain information you are adding ? such as -- server_name in nginx
-- 301 redirects nginx
-- or possibly eve route 53 to ELB mapping 

Do you know how to get the full logs from ELB ?

**Comment by hfmw** [comment]
@JungleCatSW Sorry for the delay in going back. I'm using a classic load balancer as this script doesn't work currently. Yep, tried on a fresh install and kept getting 2018/03/27 11:59:13 [emerg] 4828#0: bind() to 0.0.0.0:80 failed (98: Address already in use).

Does it work for you?? Would be great if I could get it working as an application load balancer! Cheers
